src/core/tokenizer.c

```diff
@@ -149,31 +154,83 @@ tokenset_resolve(tokenset_t *tset, nxs_index_t *idx, bool stage)
 	}
 }
 
+
+void		char_from_uchar_slice(char *, UChar* , int32_t , int32_t);
+void
+char_from_uchar_slice(char* buf, UChar* str, int32_t start, int32_t end)
+{
+	UChar   savedEndChar;
+
+	savedEndChar = str[end];
+	str[end] = 0;
+	u_austrcpy(buf, str+start);
```

Comment:
I think you want to use [u_austrncpy](https://unicode-org.github.io/icu-docs/apidoc/released/icu4c/ustring_8h.html#aa32fda5cc7663bb19b90eb792a27d46c) and then NUL terminate it. That's kind of the common way.